add ServiceCard component

InsaneZein opened the pull request:
[RHCLOUD-31482](https://issues.redhat.com/browse/RHCLOUD-31482) Adds Service Card component

![image](https://github.com/user-attachments/assets/2e0ef16c-f016-47d3-a0bd-97c4599a40d7)

patternfly-build commented:
Preview: https://react-component-groups-pr-component-groups-200.surge.sh

fhlavac commented on packages/module/patternfly-docs/content/extensions/component-groups/examples/ServiceCard/ServiceCard.md:
````diff
@@ -0,0 +1,28 @@
+import { EllipsisVIcon } from '@patternfly/react-icons';
+import contentHeaderIcon from '../../assets/icons/content-header-icon.svg'
+
+A **Service Card** component allows
+
+## Examples
+
+### Service Card
+
+```js file="./ServiceCardExample.tsx"
+
+```
+
+### Service Card with Gallery example
+
+```js file="./ServiceCardGalleryExample.tsx"
+
+```
````
@InsaneZein can you please add some texts to the docs? It would also be nice to create an example with the custom footer. Let me know if you need help with the texts.

fhlavac commented on packages/module/patternfly-docs/content/extensions/component-groups/examples/ServiceCard/ServiceCardExample.tsx:
```diff
@@ -0,0 +1,16 @@
+    description='This is a basic ServiceCard Example'
+    icon={<img src={contentHeaderIcon} alt="content-header-icon" />}
+    showDisabledButton={false}
+    helperText=''
```
I would either provide some text or remove the prop from this example

fhlavac commented on packages/module/src/ServiceCard/ServiceCard.tsx:
```diff
@@ -0,0 +1,102 @@
+
+const useStyles = createUseStyles({
+  card: {
+    height: ('var(--pf-v5-u-h-100)')
```
```suggestion
    height: 'var(--pf-v5-u-h-100)'
```

fhlavac commented on packages/module/src/ServiceCard/ServiceCard.tsx:
```diff
@@ -0,0 +1,102 @@
+    height: ('var(--pf-v5-u-h-100)')
+  },
+  image: {
+    marginRight: ('var(--pf-v5-global--spacer--md)'),
```
```suggestion
    marginRight: 'var(--pf-v5-global--spacer--md)',
```

fhlavac commented on packages/module/src/ServiceCard/ServiceCard.tsx:
```diff
@@ -0,0 +1,102 @@
+    width: 48
+  },
+  launchButton: {
+    marginRight: ('var(--pf-v5-global--spacer--sm)')
```
```suggestion
    marginRight: 'var(--pf-v5-global--spacer--md)'
```
I think this one should be `md`

fhlavac commented on packages/module/src/ServiceCard/ServiceCard.tsx:
```diff
@@ -0,0 +1,102 @@
+            ( <>
+              { launchUrl &&  
+                <Button
+                  variant={ButtonVariant.secondary}
```
This should be a primary button according to the dev sandbox implementation

fhlavac commented on packages/module/patternfly-docs/content/extensions/component-groups/examples/ServiceCard/ServiceCardGalleryExample.tsx:
Can you please add icons to these cards as well and make button disabled in one card with a helper text? Just to show the features

fhlavac commented on packages/module/src/ServiceCard/ServiceCard.test.tsx:
We should preferably use cypress

fhlavac commented on packages/module/src/ServiceCard/ServiceCard.tsx:
```diff
@@ -0,0 +1,102 @@
+import { createUseStyles } from 'react-jss';
+
+export interface ServiceCardProps {
+  /** Title for card */
```
```suggestion
  /** Service card title */
```

fhlavac commented on packages/module/src/ServiceCard/ServiceCard.tsx:
```diff
@@ -0,0 +1,102 @@
+import { HelperTextItem } from '@patternfly/react-core/dist/dynamic/components/HelperText';
+import { createUseStyles } from 'react-jss';
+
+export interface ServiceCardProps {
+  /** Title for card */
+  title: string;
+  /** Subtitle for card */
+  subtitle: string;
+  /** Custom description */
+  description: string;
+  /** icon for service card */
+  icon: React.ReactNode;
+  /** Whether to show if button is disabled */
+  showDisabledButton?: boolean;
+  /** Helper text for card */
+  helperText?: string;
+  /** learn more button url*/
+  learnMoreUrl: string;
+  /** Optional launch button url*/
+  launchUrl?: string;
+  /** Optional foot to override default Learn More and Launch buttons */
+  footer?: React.ReactElement
+  /** Optional custom OUIA ID */
+  ouiaId?: string | number;
+}
```
```suggestion
export interface ServiceCardProps {
  /** Service card title */
  title: string;
  /** Service card subtitle */
  subtitle: string;
  /** Service card description */
  description: string;
  /** Service icon */
  icon: React.ReactNode;
  /** Whether to show if button is disabled */
  showDisabledButton?: boolean;
  /** Service card helper text */
  helperText?: string;
  /** Learn more button URL */
  learnMoreUrl: string;
  /** Optional launch button URL */
  launchUrl?: string;
  /** Custom footer to override default "Learn more" and "Launch" buttons */
  footer?: React.ReactElement
  /** Optional custom OUIA ID */
  ouiaId?: string | number;
}
```

fhlavac commented on packages/module/src/ServiceCard/ServiceCard.tsx:
It would be nice to add OUIA Ids to the key component parts derived from the passed OUIA id - please see other components

fhlavac commented:
@InsaneZein thank you! Overall looks good. Just some minor comments

fhlavac commented on packages/module/patternfly-docs/content/extensions/component-groups/examples/ServiceCard/ServiceCard.md:
```diff
@@ -0,0 +1,33 @@
+import { EllipsisVIcon } from '@patternfly/react-icons';
+import contentHeaderIcon from '../../assets/icons/content-header-icon.svg'
+
+A **Service Card** component displays a card with an icon, title, description, and an optional customized footer
```
```suggestion
The **service card** component displays a card representing a service with an icon, title, description, and an optional customized footer
```

fhlavac commented on packages/module/patternfly-docs/content/extensions/component-groups/examples/ServiceCard/ServiceCard.md:
```diff
@@ -0,0 +1,33 @@
+
+### Service Card with Gallery example
+
+This shows how cards will look side by side in a `Gallery` component
```
```suggestion
This shows how cards can look side by side in a [gallery layout]((/layouts/gallery))
```

fhlavac commented on packages/module/src/ServiceCard/ServiceCard.tsx:
```diff
@@ -0,0 +1,73 @@
+  description,
+  icon,
+  helperText,
+  footer,
```
```suggestion
  footer = null,
```

fhlavac commented on packages/module/src/ServiceCard/ServiceCard.tsx:
```diff
@@ -0,0 +1,73 @@
+            </HelperTextItem>
+          </HelperText>) : null
+        }
+        { footer ? ( footer ) : null }
```
```suggestion
        { footer }
```

fhlavac commented on packages/module/src/ServiceCard/ServiceCard.tsx:
```diff
@@ -0,0 +1,73 @@
+  /** Service card icon */
+  icon: React.ReactNode;
+  /** Whether to show if button is disabled */
+  showDisabledButton?: boolean;
```
This prop is probably not needed anymore

fhlavac commented on packages/module/src/ServiceCard/ServiceCard.tsx:
```diff
@@ -0,0 +1,73 @@
+  /** Optional Service card helper text*/
+  helperText?: string;
+  /** Optional footer */
+  footer?: React.ReactElement
```
```suggestion
  footer?: React.ReactElement;
```

fhlavac commented:
The disable prop seems to be still present in the examples. Could you please also add the ouiaIds? Let's say for title, description, helper text and footer. Thank you

InsaneZein commented:
> The disable prop seems to be still present in the examples. Could you please also add the ouiaIds? Let's say for title, description, helper text and footer. Thank you

Oops, fixed that and added the ouiaIds

fhlavac commented:
perfect, thanks @InsaneZein.
Just the cypress tests and we are golden! 🏅

InsaneZein commented:
@fhlavac added a couple cypress tests
